### DISCUSSION TITLE
Trigger a workflow from another workflow

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#26294] [Actions]
<p>I remember seeing somewhere that you could trigger a different workflow from another workflow?</p>
<p>For example, if Workflow A is successful would it trigger Workflow B to send a notification about the success.</p>
<p>Is this still possible, or did something change?</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Yep, you can <strong><a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#external-events-repository_dispatch" rel="nofollow noopener">c</a>****<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#external-events-repository_dispatch" rel="nofollow noopener">reate a repository dispatch event</a></strong> to trigger a workflow from another workflow.</p>
<p>For example, there 2 workflows ( <strong>wf1</strong> and <strong>wf2</strong> ), they can be in a same repository or different repositories. At the end of wf1, you can add an additional job to create a repository dispatch event via executing the <strong><a href="https://developer.github.com/v3/repos/#create-a-repository-dispatch-event" rel="nofollow noopener">GitHub REST API</a></strong> for the repository that cotains the wf2. And in wf2, you should set its trigger event as  <strong>on: repository_dispatch</strong></p>
<p>A simple example:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6d8217ef3f7aa4bdadf2bbe433f41b4c85f79271.png" data-download-href="/uploads/short-url/fCKLDmDPZ782PBGH4aFUctzROw1.png?dl=1" title="example.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181096709-c932bbb3-2419-4684-b4ac-877a3d29584f.png" alt="example.PNG" data-base62-sha1="fCKLDmDPZ782PBGH4aFUctzROw1" width="634" height="499" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6d8217ef3f7aa4bdadf2bbe433f41b4c85f79271_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">example.PNG</span><span class="informations">739×582 15.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>NOTE:</strong> you need to create a new PAT instead of the automatically created <strong>GITHUB_TOKEN</strong> to authenticate the API. This new PAT should have permissions to access the repository that cotains wf2, and you aslo need to add this new PAT as a secret in the repository that cotains wf1.</p>

**Reply by marcodicro-dp** [reply]
How is this different from the `on: workflow_call:`?

**Reply by shalva97** [reply]
Doing a API request in Windows terminal, using .NET APIs, which is written in YML.... Good luck maintaining it

**Comment by peter-evans** [comment]
<p><a href="https://github.com/peter-evans/repository-dispatch" rel="nofollow noopener">This action</a> might help you if you want to use an <code>repository_dispatch</code> event workflow to achieve this.</p>

**Reply by zpostfacto** [reply]
THANK YOU

**Comment by Andre601** [comment]
<p>So I could also use this across repositories? That’s handy, as I would like to automatically update specific files in a repository when I update them on another.</p>
<p><a href="https://github.community/t/is-there-a-way-to-copy-content-from-one-repository-to-another-automatically/2843">I asked about this here</a> but didn’t get an answer yet. So if you have an idea of how I could achieve this, let me know.</p>

**Comment by DanielTillmannTR** [comment]
<p>Can this be achieved by using GITHUB_TOKEN yet?</p>
<p>I have security requirements to no longer use PATs.</p>

**Comment by rohan-sca** [comment]
<p>Same requirement on my end. can this be done with the GITHUB_TOKEN instead of a PAT? or perhaps some other mechanism which has a limited security concern Vs the PAT ?</p>

**Comment by ScottGuymer** [comment]
<p>We have the same drive to move away from PAT tokens as they are difficult to scope well and are tied to a single user (not to mention using up a paid user seat in your org)</p>
<p>You can do this with a GH App by using this action to generate a token and then using it in subsequent steps <a href="https://github.com/philips-software/app-token-action" class="inline-onebox" rel="noopener nofollow ugc">GitHub - philips-software/app-token-action: GitHub action to obatin app token</a></p>
<p>You need to make sure you grant the app the correct permissions on your repo (or in your org) and this works well with org level apps.</p>

**Comment by whsalazar** [comment]
It is now possible to do this with GITHUB_TOKEN - https://github.blog/changelog/2022-09-08-github-actions-use-github_token-with-workflow_dispatch-and-repository_dispatch/

**Reply by epoctic** [reply]
Important note: the GITHUB_TOKEN appears to only enable dispatching workflows that exist in the same repository.

**Reply by Andre601** [reply]
That makes sense since the token is to my knowledge scoped to the repo itself.
